src/app/utils/helper.ts

```diff
@@ -0,0 +1,235 @@
+
+  // Create a FileReader to read the Blob as a data URL
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+
+    // Set up the onload event to resolve the promise with the data URL
+    reader.onloadend = () => resolve(reader.result);
+
+    // Set up the onerror event to reject the promise
+    reader.onerror = () => reject('Failed to convert blob to data URL');
+
+    // Read the Blob as a data URL
+    reader.readAsDataURL(blob);
+  });
+}
+
+export const toCSS = async function (js: string): Promise<string> {
+  try {
+    return new Promise((resolve, reject) => {
+      let val: CssInJs = {};
+      eval(`val = ${js}`);
+      postcss()
+        // @ts-ignore
+        .process(val, { parser: cssjs })
+        .then((code) => resolve(code.css))
+        .catch((err) => reject(err));
+    });
+  } catch (err) {
+    console.error("Couldn't convert JS to CSS.", err);
+    return '';
+  }
+};
+
+export const generateHtml = async (document: MockDocumentDataType) => {
+  const dataToConsider = document;
+  const documentName = dataToConsider.name;
+  const sectionData: any = dataToConsider.sections;
+
+  let headerGlobalCss: string = '';
+  let globalFontFamily: string = '';
+  let globalFontSize: string = '';
+
+  if (dataToConsider.styles.GLOBAL.backgroundColor) {
+    const backgroundColorJson = {
+      backgroundColor: dataToConsider.styles.GLOBAL.backgroundColor,
+    };
+
+    headerGlobalCss = await toCSS(JSON.stringify(backgroundColorJson));
+  }
+
+  if (dataToConsider.styles.GLOBAL.fontFamily) {
+    const fontFamilyJson = {
+      fontFamily: dataToConsider.styles.GLOBAL.fontFamily,
+    };
+
+    globalFontFamily = await toCSS(JSON.stringify(fontFamilyJson));
+  }
+
+  if (dataToConsider.styles.GLOBAL.fontSize) {
+    const fontSizeJson = {
+      fontSize: dataToConsider.styles.GLOBAL.fontSize,
+    };
+
+    globalFontSize = await toCSS(JSON.stringify(fontSizeJson));
+  }
+
+  let htmlData: string = '';
+  let headerHtml = '';
+  let firstColumnHtml: string[] = [];
+  let secondColumnHtml: string[] = [];
+  let imageUrl = '';
+
+  // iterate over the sections keys
+  const keys: string[] = Object.keys(sectionData);
+  keys.forEach((key: string) => {
+    const sectionDetails = sectionData[key];
+    const currSectioName = sectionDetails.sectionName;
+
+    const currSectionSchema = sectionMockJson.find(
+      (section) => section.name === currSectioName
+    );
+    if (!currSectionSchema) return;
+
+    // we also need to remove the <img tag from the titleComp as we'll be adding it via jspdf
+    currSectionSchema.titleComp = currSectionSchema.titleComp.replace(
+      /<img[^>]*>/g,
+      ''
+    );
+
+    // iterate over the keys to replace
+    allKeysToReplace.forEach((keyToReplace) => {
+      const keyToReplaceRegex = new RegExp('\\' + keyToReplace, 'g');
+
+      const keyToReplaceValue =
+        sectionDetails.body[0][keyToReplace.split('${this.')[1].split('}')[0]];
```

Comment:
why are you using the `[0]` index of body?
A sections body can have multiple items (all would reference the same list of keys)

Reply:
<img width="365" alt="image" src="https://github.com/dompainter/pdf-generator/assets/17257060/cadf1bbf-8146-4cbf-90a0-a96b8d33f39f">
In this example, experience has 2 "body" items. 1 being the google job and the other being canva. We need to be able to support multiple items being rendered in here

Reply:
Fixed.